app/src/main/java/com/willowtree/vocable/keyboard/KeyboardFragment.kt

```diff
@@ -104,10 +108,12 @@ class KeyboardFragment : BaseFragment<FragmentKeyboardBinding>() {
 
         binding.keyboardSpeakButton.action = {
             if (!isDefaultTextVisible()) {
+                val text = binding.keyboardInput.text?.toString() ?: ""
                 VocableTextToSpeech.speak(
                     Locale.getDefault(),
-                    binding.keyboardInput.text?.toString() ?: ""
+                    text
                 )
+                viewModel.addNewPhrase(text, PresetCategories.RECENTS.id)
```

Comment:
You exposed a flaw in how our categories are set up, nice! We don't actually want to be storing phrases under the "RECENTS" category, instead we determine RECENTS by just looking at the most recently used phrases [PhrasesUseCase#getPhrasesForCategory()](https://github.com/willowtreeapps/vocable-android/blob/492d1409be508a3c6ecd81181985da3d0a75879a/app/src/main/java/com/willowtree/vocable/PhrasesUseCase.kt#L23).

Totally understand how you got here, and we should clean up the API to not allow creating a phrase with "Recent" as the category, could you create a ticket on the board to track that? Up to you whether or not you tackle that, but we should get it written down.

As far as the defect at hand, I don't think we actually want just anything spoken from the keyboard to be saved, and this is corroborated by how the iOS app works. Instead, Android is currently missing functionality where a user can save a phrase directly from the keyboard. Download the Vocable iOS app to your Mac and poke around to see what I mean. Once you feel comfortable, I think we should discard the current bug as "won't-fix" and write up an enhancement to match iOS's functionality here.

Reply:
Updated #445 to reflect adding the My Sayings enhancement.

Reply:
@cinadia Agreed!